indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/ParallelIndexSupervisorTask.java

```diff
@@ -555,14 +559,23 @@ private TaskStatus runSinglePhaseParallel(TaskToolbox toolbox) throws Exception
     );
 
     final TaskState state = runNextPhase(runner);
+    TaskStatus taskStatus;
     if (state.isSuccess()) {
       //noinspection ConstantConditions
       publishSegments(toolbox, runner.getReports());
       if (awaitSegmentAvailabilityTimeoutMillis > 0) {
         waitForSegmentAvailability(runner.getReports());
       }
+      taskStatus = TaskStatus.success(getId());
+    } else {
+      // there is only success or failure after running....
+      Preconditions.checkState(state.isFailure());
```

Comment:
Is this a future proof for any new states that can be transitioned from running besides success and failure? Could it be nice to include some messages about what the state was?

Reply:
Today there is no state other than failure after a task ran and was not successful. I added a log message for the case when that state is added so we know that there is some code that needs to be changed as well as throwing an exception to let the person change the code in an appropriate manner for the new state.